org.testeditor.web.dropwizard.xtext/src/main/java/org/testeditor/web/xtext/index/changes/GradleBuildChangeDetector.xtend

```diff
@@ -50,18 +50,25 @@ class GradleBuildChangeDetector implements ChangeDetector {
 	var projectRoot = memoize[new File(config.get.localRepoFileRoot)]
 	var lastDetectedResources = <URI>emptySet
 
-	override detectChanges(ResourceSet resourceSet, String[] paths, ChangedResources accumulatedChanges) {
+	override ChangedResources collectFull(ResourceSet resourceSet, String[] paths, ChangedResources accumulatedChanges) {
```

Comment:
similar to the `GitBasedChangeDetector`, `collectFull` seems to be a special case of `detectChanges` for when `lastDetectResources` is empty. So, as an alternative to the boolean flag `detectDeleted`, you could also just reset that set.